core/api/serializers/cp_record.py

```diff
@@ -98,6 +98,18 @@ class Meta(CPRecordBaseSerializer.Meta):
         model = CPRecord
 
 
+class CPRecordReadOnlySerializer(CPRecordBaseSerializer):
+    record_usages = serializers.SerializerMethodField()
+
+    def get_record_usages(self, obj):
+        if obj.id == 0:
+            return []
+        return CPUsageSerializer(obj.record_usages.all(), many=True).data
+
+    class Meta(CPRecordBaseSerializer.Meta):
+        model = CPRecord
```

Comment:
This serializer was necessary because in the `CPRecordSerializer` `record_usages` ends up making a query for each `CPRecord` created with id=0 in `_set_chemical_items_dict`. I couldn't set the reverse relationship `record_usages` when creating the objects.

`CPRecord`s with id=0 are not used so their `record_usages` would be an empty list (as should be the result for each `CPUsage` query for these records).

If the record comes from the db and has an id, the usages are prefetched so the serializer doesn't need extra queries.

I couldn't think of another way, so suggestions are more than welcome.

Reply:
@andreighh good idea! I don't think there's a way to prefetch reverse relationships on objects that are actually being created. :)